pkg/tsdb/testdatasource/scenarios.go

```diff
@@ -848,14 +925,22 @@ func predictablePulse(query backend.DataQuery, model *simplejson.Json) (*data.Fr
 	frame := newSeriesForQuery(query, model, 0)
 	frame.Fields = fields
 	frame.Fields[1].Labels = parseLabels(model)
-
+	addDsFieldConfig(frame.Fields[1], model)
 	return frame, nil
 }
 
-func randomHeatmapData(query backend.DataQuery, fnBucketGen func(index int) float64) *data.Frame {
+func randomHeatmapData(query backend.DataQuery, fnBucketGen func(index int) float64) (*data.Frame, error) {
+	model, err := simplejson.NewJson(query.JSON)
+
+	if err != nil {
+		return nil, fmt.Errorf("error reading query")
```

Comment:
The original error should be kept, so the context isn't lost.
```suggestion
		return nil, fmt.Errorf("error reading query: %w", err)
```